packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -457,14 +460,16 @@ export class ProjectModel {
     async saveWarehouseToCache(
         projectUuid: string,
         warehouse: WarehouseCatalog,
-    ): Promise<{ projectUuid: string; warehouse: WarehouseCatalog }> {
-        return this.database(CachedWarehouseTableName)
+    ): Promise<DbCachedWarehouse> {
+        const [cachedWarehouse] = await this.database(CachedWarehouseTableName)
             .insert({
                 project_uuid: projectUuid,
                 warehouse: JSON.stringify(warehouse),
             })
             .onConflict('project_uuid')
             .merge()
-            .select(['project_uuid', 'warehouse']);
+            .returning('*');
+
+        return cachedWarehouse;
```

Comment:
same here, `cachedWarehouse` could be `undefined` - it's annoying that typescript doesn't figure this out for us